services/w3c/w3c-validation.service.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+const Joi = require('@hapi/joi')
+const { optionalUrl } = require('../validators')
+const {
+  documentation,
+  presetRegex,
+  getColor,
+  getMessage,
+  getSchema,
+} = require('./w3c-validation-helper')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  messages: Joi.array()
+    .required()
+    .items(Joi.object()),
+}).required()
+
+const presetDecode = regExpression => (preset, helpers) => {
+  if (!preset) {
+    return ''
+  }
+
+  if (regExpression && !regExpression.test(decodeURI(preset))) {
+    return helpers.error('regex')
+  }
+
+  // Return the value unchanged
+  return preset
+}
+
+const queryParamSchema = Joi.object({
+  targetUrl: optionalUrl.required(),
+  preset: Joi.string()
+    .allow('')
+    .custom(presetDecode(presetRegex), 'preset validation'),
```

Comment:
Is it necessary to use a custom validator here? 
Would something like the below work? Or is the custom validator required due to the query params being uri encoded?
```js
preset: Joi.string().regex(presetRegex).allow(null),

Reply:
Using the regex validator but doing `allow('')` instead of `allow(null)`